### PR TITLE
Preview different Plan UI's with a radio toggle

### PR DESCRIPTION
**Pull request by dereksmart**
This PR adds a new component `DevCard` to the mix, which is rendered in the footer only if Jetpack is running in a **development version**.  

Right now it only allows you to toggle between different Jetpack plans so that you're able to view the **UI** differences.  

_**Toggling between these is not actually changing your plan, it is only updating the state for viewing/testing purposes of the UI**_ -- aka: Your plan is expected to revert back to your actual plan when the /site endpoint is queried next, so that is NOT a bug.  

There is no doubt that this can be improved/expanded a great deal in the future (and it will), but for now **this PR is meant to be a quick win for designers/growth when they are testing differences between plans**. 

**To Test:** 
- Checkout and view the `/wp-admin/admin.php?page=jetpack#/plans` tab.  
- Toggle between the plans.  The view should reflect accordingly.  
- Navigate to the dashboard.  Your plan should be updated along with the /site endpoint response.  
- Change the Jetpack version in jetpack.php to `4.8`, make sure the component does not render.  

switchplan

### CODE REVIEW

File: _inc/client/components/dev-card/index.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { isDevVersion as _isDevVersion } from 'state/initial-state';
+import { switchMichaelsPlan } from 'state/dev-version';
```
Comment (samhotchkiss) [inline _inc/client/components/dev-card/index.jsx]: Let's not use people's names in our codebase?

File: _inc/client/state/dev-version/actions.js [hunk 2]
```diff
@@ -35,4 +36,13 @@ export const resetOptions = ( options ) => {
 			dispatch( createNotice( 'is-error', __( 'Options failed to reset.' ), { id: 'reset-options' } ) );
 		} );
 	}
-}
+};
+
+export const switchMichaelsPlan = ( slug ) => {
```
Comment (samhotchkiss) [inline _inc/client/state/dev-version/actions.js]: How about "previewAnotherPlan" or something like that?

**Comment by samhotchkiss** [comment]
Noice!

Should we just be showing this to all users running a dev version?  Our goal over the next few months is to get many more beta testers, and this seems like it might be a bit unnecessary/confusing.  What about a little footer link to turn this thing on?  Or...

**Comment by dereksmart** [comment]
@samhotchkiss I was thinking of putting it behind a filter

**Comment by dereksmart** [comment]
I've updated this to add state/actions to be able to toggle the devcard.  I've added a link to toggle it specifically in the footer in dev versions only, similar to how we have `reset options`.  

This is what it looks like 
switchplan-toggle

**Comment by samhotchkiss** [comment]
Love it!

**Pull request merged**